File: src/main/java/net/igneo/icv/entity/chestplate/soulSpider/SoulSpiderModel.java
package net.igneo.icv.entity.chestplate.soulSpider;

import net.igneo.icv.ICV;
import net.minecraft.resources.ResourceLocation;
import software.bernie.geckolib.model.GeoModel;

public class SoulSpiderModel extends GeoModel<SoulSpiderEntity> {
    private final ResourceLocation model = new ResourceLocation(ICV.MOD_ID, "geo/soul_spider.geo.json");
    private final ResourceLocation texture = new ResourceLocation(ICV.MOD_ID, "textures/entity/soul_spider.png");
    private final ResourceLocation animations = new ResourceLocation(ICV.MOD_ID, "animations/soul_spider.animation.json");
    
    
    @Override
    public ResourceLocation getModelResource(SoulSpiderEntity soulSpiderEntity) {
        return this.model;
    }
    
    @Override
    public ResourceLocation getTextureResource(SoulSpiderEntity soulSpiderEntity) {
        return this.texture;
    }
    
    @Override
    public ResourceLocation getAnimationResource(SoulSpiderEntity soulSpiderEntity) {
        return this.animations;
    }
}

File: src/main/java/net/igneo/icv/entity/chestplate/soulSpider/SoulSpiderModel.kt
package net.igneo.icv.entity.chestplate.soulSpider

import net.igneo.icv.ICV
import net.minecraft.resources.ResourceLocation
import software.bernie.geckolib.model.GeoModel

class SoulSpiderModel : GeoModel<SoulSpiderEntity?>() {
    private val model = ResourceLocation(ICV.MOD_ID, "geo/soul_spider.geo.json")
    private val texture = ResourceLocation(ICV.MOD_ID, "textures/entity/soul_spider.png")
    private val animations = ResourceLocation(ICV.MOD_ID, "animations/soul_spider.animation.json")


    override fun getModelResource(soulSpiderEntity: SoulSpiderEntity?): ResourceLocation {
        return this.model
    }

    override fun getTextureResource(soulSpiderEntity: SoulSpiderEntity?): ResourceLocation {
        return this.texture
    }

    override fun getAnimationResource(soulSpiderEntity: SoulSpiderEntity?): ResourceLocation {
        return this.animations
    }
}
